libsolidity/interface/StandardCompiler.cpp

```diff
@@ -730,7 +731,26 @@ std::variant<StandardCompiler::InputsAndSettings, Json::Value> StandardCompiler:
 		for (auto const& sourceName: sources.getMemberNames())
 			ret.sources[sourceName] = util::jsonCompactPrint(sources[sourceName]);
 	}
+	else if (ret.language == "EVMAssembly")
+	{
+		for (std::string const& sourceName: sources.getMemberNames())
+		{
+			solAssert(sources.isMember(sourceName));
+			if (!sources[sourceName].isMember("assemblyJson") ||
+				!sources[sourceName]["assemblyJson"].isObject() || sources[sourceName].size() != 1)
+				return formatFatalError(
+					Error::Type::JSONError,
+					"Invalid input source specified. Expected exactly one 'assemblyJson' object inside $.sources." + sourceName
```

Comment:
This will make the message more understandable for cases where there the error is caused by other keys being included in the dict:
```suggestion
					"Invalid input source specified. Expected exactly one key, named 'assemblyJson' object inside $.sources." + sourceName
```